src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1347,6 +1350,16 @@ void BlockchainLMDB::open(const std::string& filename, const int db_flags)
     throw DB_ERROR("Database could not be opened");
   }
 
+#ifdef WIN32
+  // ensure NTFS compression is disabled on the directory and database file to avoid corruption of the blockchain 
+  if (boost::filesystem::exists(filename) && !disable_ntfs_compression(filename)) 
```

Comment:
The earlier logic creates the directory (~line 1337), so you shouldn't need to check for existence (the extra check doesn't add race protection, and you don't need race protection).